gh-1123 document next, ipairs, upsert tuple functions (#1148)
tarantool/doc: doc/reference/reference_lua/box_tuple.rst

    | :ref:`tuple_object:findall()         | Get the number of all fields    |
    | <box_tuple-find>`                    | matching the search value       |
    +--------------------------------------+---------------------------------+

>     | :ref:`tuple_object:next()            | Get the next field value from a     |

Reviewer: @Onvember table frame is invalid

            - ['Fld#1', 'B', 'Fld#3', 'Fld#4', 'Fld#5']
            ...

    .. _box_tuple-upsert:

    .. method:: upsert({{operator, field_no, value}, ...})

        The same as ``tuple_object:update()``, but ignores errors. In case
        of an error the tuple is left intact, but an error message is
        printed. Only client errors are ignored, such as a bad field type,
        or wrong field index/name. System errors, such as OOM, are not

>         ignored and raised just like with normal ``update()``. Note, that

Reviewer: ```suggestion
        ignored and raised just like with a normal ``update()``. Note that
```

        only bad operations are ignored. All correct operations are
        applied.

>         :param string  operator: operation type represented in string (e.g.

Reviewer: ```suggestion
        :param string  operator: operation type represented as a string (e.g.
```

                                 '``=``' for 'assign new value')

>         :param number  field_no: what field the operation will apply to. The

Reviewer: ```suggestion
        :param number  field_no: the field to which the operation will be applied. The
```

                                 field number can be negative, meaning the
                                 position from the end of tuple.
                                 (#tuple + negative field number + 1)

>         :param lua_value  value: what value will be applied

Reviewer: ```suggestion
        :param lua_value  value: the value which will be applied
```

        :return: new tuple
        :rtype:  tuple

>         See how in the next example one operation is applied, and one is not.

Reviewer: ```suggestion
        See the following example where one operation is applied, and one is not.
```
